include/drivers/flash.h

```diff
@@ -224,6 +215,9 @@ static inline int z_impl_flash_erase(const struct device *dev, off_t offset,
 /**
  *  @brief  Enable or disable write protection for a flash memory
  *
+ *  This API is deprecated and will be removed in Zephyr 2.8.
+ *  It will be keep as No-Operation until that.
```

Comment:
This too should mention that the responsibility for this function is now transferred to the write and erase operations.